### ISSUE TITLE
Upload all existing logos from Whitehall to Asset Manager

### ISSUE THREAD

**Issue by chrisroos** [alphagov/asset-manager#276]
- [x] Write [some code](https://github.com/alphagov/whitehall/pull/3513) to upload all existing logos
- [x] Run `asset_manager:migrate_organisation_logos` on Integration
- [x] Merge https://github.com/alphagov/whitehall/pull/3515
- [x] Deploy https://github.com/alphagov/whitehall/pull/3515 to integration
- [x] Run `asset_manager:migrate_organisation_logos` on Integration
- [x] Compare the response headers on integration when requesting an organisation logo from Whitehall versus Asset Manager
- [x] Deploy https://github.com/alphagov/whitehall/pull/3515 to staging
- [x] Deploy https://github.com/alphagov/whitehall/pull/3515 to production
- [x] Run `asset_manager:migrate_organisation_logos` on Staging
- [x] Run `asset_manager:migrate_organisation_logos` on Production

**Comment by chrislo** [comment]
I've merged https://github.com/alphagov/whitehall/pull/3513. We now need to run the rake task `asset_manager:migrate_organisation_logos` on all environments.

**Comment by chrislo** [comment]
I'm [running the task in integration](https://deploy.integration.publishing.service.gov.uk/job/run-rake-task/3404/console)

**Comment by chrislo** [comment]
The task has completed. I saw jobs being added to the `asset_migration` queue [in Grafana](https://grafana.integration.publishing.service.gov.uk/dashboard/db/sidekiq-copy?refresh=5s&orgId=1&from=now-7d&to=now&var-Application=whitehall&var-Queues=asset_migration) but the queue was drained too quickly to get a useful screenshot. Running the following command in the `asset-manager` Rails console on Integration:

```
WhitehallAsset.pluck(:legacy_url_path).grep(/organisation\/logo/).size #=> 106
```

Shows that 106 assets have been uploaded. This agrees with the number of files in the NFS mount 

```
$ find /data/uploads/whitehall/clean/system/uploads/organisation/logo/ -type f | wc -l 
106
```

I'm fairly confident we can run this task on Staging and Production now.

**Comment by chrisroos** [comment]
I've been comparing the response headers between the organisation logos being served by Asset Manager and those being served by Whitehall. There are a couple of differences that I'm investigating so we can't yet run this Rake task in staging and production.

**Comment by chrisroos** [comment]
I've opened PR #3515 to fix the discrepancies in the response headers when requesting the same file from Whitehall and from Asset Manager.

**Comment by chrisroos** [comment]
I've merged https://github.com/alphagov/whitehall/pull/3515 so we'll be in a position to run the `asset_manager:migrate_organisation_logos` task again in the morning.

**Comment by chrisroos** [comment]
I've run the Rake task on integration and compared the response headers from Asset Manager and Whitehall. You can see that the `Last-Modified`, `ETag` and `Content-Length` are the same in both cases, which gives me confidence that the migration was successful.

## Requesting a Whitehall organisation logo from Asset Manager

```
$ curl https://asset-manager.integration.publishing.service.gov.uk/government/uploads/system/uploads/organisation/logo/723/NPS_logo.png?proxy_to_s3_via_nginx=true -v > /dev/null

< HTTP/1.1 200 OK
* Server nginx is not blacklisted
< Server: nginx
< Date: Wed, 01 Nov 2017 09:50:16 GMT
< Content-Type: image/png
< Content-Length: 12981
< Connection: close
< Content-Disposition: inline; filename="NPS_logo.png"
< Cache-Control: max-age=14400, public
< Last-Modified: Mon, 02 Jun 2014 11:44:14 GMT
< ETag: "538c638e-32b5"
< x-amz-meta-md5-hexdigest: 8f460cebb6905b1ec07ad1f40e929f07
< Accept-Ranges: bytes
< GOVUK-Asset-Manager-File-Store: S3
< X-Frame-Options: SAMEORIGIN
< Strict-Transport-Security: max-age=31536000
```

## Requesting a Whitehall organisation logo from Whitehall (via assets-origin)

```
$ curl https://assets-origin.integration.publishing.service.gov.uk/government/uploads/system/uploads/organisation/logo/723/NPS_logo.png -v > /dev/null

< HTTP/1.1 200 OK
* Server nginx is not blacklisted
< Server: nginx
< Date: Wed, 01 Nov 2017 09:52:12 GMT
< Content-Type: image/png
< Content-Length: 12981
< Connection: close
< Last-Modified: Mon, 02 Jun 2014 11:44:14 GMT
< Content-Disposition: inline; filename="NPS_logo.png"
< Cache-Control: max-age=14400, public
< ETag: "538c638e-32b5"
< X-Frame-Options: SAMEORIGIN
< Accept-Ranges: bytes
< Strict-Transport-Security: max-age=31536000
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET, OPTIONS
< Access-Control-Allow-Headers: origin, authorization
```

**Comment by chrisroos** [comment]
I think we're in a position to request that Whitehall is deployed and this task is run on staging and production.

**Comment by chrislo** [comment]
With the help of @pmanrubia we ran this on Staging and then Production today. There were some issues on Staging due to a mis-configured API token which Whitehall required to authenticate with Asset Manager. We generated a new token and added it to `govuk_secrets`. We then ran the rake task on Staging and checked the result in Asset Manager

```
irb(main):001:0> WhitehallAsset.pluck(:legacy_url_path).grep(/organisation\/logo/).size
=> 106
```

And the same against Production 

```
WhitehallAsset.pluck(:legacy_url_path).grep(/organisation\/logo/).size
=> 106
```

This ties with the 106 files under `/organisation/logo` that we saw in the NFS mount replicated to Integration.
